packages/client-core/src/admin/components/Project/ProjectFields.tsx

```diff
@@ -106,7 +106,7 @@ const ProjectFields = ({ inputProject, existingProject = false, changeDestinatio
     try {
       ProjectUpdateService.resetSourceState(project, { resetSourceURL: false })
       ProjectUpdateService.setBranchProcessing(project, true)
-      const branchResponse = (await ProjectService.fetchProjectBranches(e.target.value)) as any
+      const branchResponse = (await Engine.instance.api.service('project-branches').get(e.target.value)) as any
```

Comment:
Is there any discussion as to how we handle these service calls on handlers? Maybe we can have a `useQuery` hook which will some what work like `useMutation` but it will return get and find methods?

Reply:
we already have `useQuery`, `useGet`, and `useFind` hooks. Can't we use those ?

Reply:
Ohh okay, but I dont think there is a way to call them in a callback like above. There is only a refetch method that it offers.